### PR TITLE
Init block dragger

### PR DESCRIPTION
**Pull request by alschmiedt**
- Adds the block drag surface to the registry
- Adds `moveBlockWhileDragging` to the block dragger
- Adds `translateBy` to the block drag surface

### CODE REVIEW

File: core/block_dragger.js
Comment (maribethb) [inline core/block_dragger.js:226]: I think it would be helpful to add a bit more description to this method. A year from now we are going to look back on this like "what the heck is this even doing" and I think describing what we found out this quarter, and explaining the motivation behind the method and what it's doing (e.g. "When the workspace is dragged, x happens, but when a block is dragged, y happens and you need to account for z"). Maybe we should even add those details to other methods in this class. I'm also happy to work on this separately / not blocking this change. I already need to review what x, y, and z are to make sure I remember it correctly :)

File: core/interfaces/i_block_drag_surface.js
Comment (maribethb) [inline core/interfaces/i_block_drag_surface.js:77]: I think this should be `getCurrentBlock`

**Review by maribethb — APPROVED** [review]

File: core/block_dragger.js
Comment (alschmiedt) [reply core/block_dragger.js:226]: Good point and I completely agree. I definitely don't want to have to figure out the block dragger logic again. I will plan to add more documentation once I am sure where this function should live since I am not 100% convinced this is the right place for it just yet.

File: core/interfaces/i_block_drag_surface.js
Comment (alschmiedt) [reply core/interfaces/i_block_drag_surface.js]: Whoops, good catch :)

File: core/block_dragger.js
Comment (maribethb) [reply core/block_dragger.js:226]: Sounds good, we can do a documentation pass before we merge this branch to develop.

**Pull request merged**
